src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -1231,11 +1232,12 @@ void deliverReply(Message msg) {
             statistics.incrementRepliesReceived();
         }
         else if (!oldStyle && !subject.startsWith(mainInbox)) {
-            if (advancedTracking && responsesRespondedTo.get(key) != null) {
-                statistics.incrementDuplicateRepliesReceived();
-            }
-            else {
-                statistics.incrementOrphanRepliesReceived();
+            if (advancedTracking) {

```

Comment:
Previously, all orphan counter was being incremented for every duplicate or orphan when `advancedTracking` was not enabled. Does this change to first check the `advancedTracking` flag and only when enabled increment duplicate or orphan replies.